pkg/server/grpc/grpc.go

```diff
@@ -29,13 +31,14 @@ type Server struct {
 	comp   component.Component
 
 	srv      *grpc.Server
+	h        *health.Server
 	listener net.Listener
 
 	opts options
 }
 
 // New creates a new Server.
-func New(logger log.Logger, reg prometheus.Registerer, tracer opentracing.Tracer, comp component.Component, storeSrv storepb.StoreServer, opts ...Option) *Server {
+func New(logger log.Logger, reg prometheus.Registerer, tracer opentracing.Tracer, comp component.Component, h *health.Server, storeSrv storepb.StoreServer, opts ...Option) *Server {
```

Comment:
I think we should pass just interface to health Server not the pointer to exact implementation